src/main/kotlin/model/PrepareCaseForCourt.kt

```diff
@@ -1,33 +1,113 @@
+        "PostgreSQL"
+      ).apply {
+        Tags.DATABASE.addTo(this)
+        Tags.SOFTWARE_AS_A_SERVICE.addTo(this)
+      }
+
+      val messagesDb = system.addContainer(
+        "Messages Database",
+        "Holds CPMG messages data and meta-data",
+        "PostgreSQL"
+      ).apply {
+        Tags.DATABASE.addTo(this)
+        Tags.SOFTWARE_AS_A_SERVICE.addTo(this)
+      }
+
+      courtCaseService = system.addContainer(
+        "Court Case Service",
+        "Court case business logic and REST API consumed by the Prepare a Case web application",
+        "Java + Spring Boot"
+      ).apply {
+        uses(casesDb, "connects to", "JDBC")
+      }
+
+      courtCaseMatcher = system.addContainer(
+        "Court Case Matcher",
+        "Consumes court lists from Crime Portal and matches defendants  against the nDelius database via Offender Search",
+        "Java + Spring Boot"
+      ).apply {
+        uses(courtCaseService, "Creates or updates cases in")
+      }
+
+      prepareCaseUI = system.addContainer(
+        "Prepare a Case UI",
+        "Web application  which delivers  HTML to the users browser",
+        "Node + Express"
+      ).apply {
+        uses(courtCaseService, "View case defendant details and capture court judgements using ")
+      }
+
+      val crimePortalMirrorGateway = system.addContainer(
+        "Crime Portal Mirror Gateway",
+        "OAP web service which receives court lists and publishes onto a JMS message queue",
+        "JBoss Wildfly"
+      ).apply {
+        uses(messagesDb, "connects to", "JDBC")
+        uses(courtCaseMatcher, "Sends court lists to")
+      }
+
+      // TODO refactor out HMCTS Crime Portal into dedicated SoftwareSystem
+      model.addSoftwareSystem("HMCTS Crime Portal", "Case Management for HMCTS holding data relating to court cases")
+        .apply {
+          setLocation(Location.External)
+          Tags.PROVIDER.addTo(this)
+          uses(crimePortalMirrorGateway, "Sends court lists to")
+        }
     }
 
     override fun defineRelationships() {
-      CourtUsers.courtAdministrator.uses(system, "captures court judgements in")
-      ProbationPractitioners.nps.uses(system, "view case defendant details")
+      listOf(prepareCaseUI, courtCaseService, courtCaseMatcher)
+        .forEach { it.uses(HMPPSAuth.system, "authenticates via") }
+
+      courtCaseService.uses(Delius.communityApi, "Gets offender details from")
+      courtCaseMatcher.uses(Delius.offenderSearch, "Matches defendants to known offenders")
 
-      system.uses(Delius.system, "get offender details from")
-      system.uses(OASys.system, "get offender assessment details from")
+      // TODO Model OASys & link to Offender Assessment API
+      courtCaseService.uses(OASys.system, "get offender assessment details from")
+
+      CourtUsers.courtAdministrator.uses(system, "prepares cases for sentencing")
+      ProbationPractitioners.nps.uses(system, "views case defendant details")
```

Comment:
What do you think about redirecting these use cases to containers? Otherwise, the users won't be connected to anything in the container diagram (as the person&rarr;system relationship is not shown there).

Reply:
I was wondering where they had gone after the merge :man_facepalming:  Thanks for the pointers :thumbsup: